### ISSUE TITLE
What do we need to make z20 happen?

### ISSUE THREAD

**Issue by dktue** [gravitystorm/openstreetmap-carto#3686]
In different tickets we're having a side-discussion mentioning a potential zoomlevel 20 (z20). Clearly, many mappers agree that it would be beneficial for users to have another more detailed level.

# Discussion
I would like to start a discussion what would be needed to make z20 happen. I'm aware that the map-style of course itself would need to be tweaked for the new zoomlevel. But there might be other aspects like technical issues of the OSM-tile-servers not having enough capacity.

Probably this repository is not the perfect place for all related topics to z20, but I want to get a starting point and maybe links to other issues in other repositories. The goal is connect different developers to find out what needs to be done.

**What do we need to make z20 happen?**

**Comment by HolgerJeromin** [comment]
See https://github.com/openstreetmap/openstreetmap-website/issues/2094#issuecomment-450636269

**Comment by kocio-pl** [comment]
In this style we have all the zoom levels down to infinity. You probably mean OSMF servers to render z20 - this does not belong here, you need to talk directly with OSMF OWG:

https://github.com/openstreetmap/operations/issues

**Comment by dktue** [comment]
Of course, technically this style could be rendered to infinity. But probably we would like to apply some tweaks when rendering z20.

**Comment by kocio-pl** [comment]
We can think about them if OWG will agree to render z20 first, however if you have some idea what could it be, please be more specific.

**Comment by kocio-pl** [comment]
In fact our default settings are z-max=22:

https://github.com/gravitystorm/openstreetmap-carto/blob/master/project.mml#L17

**Comment by Adamant36** [comment]
@kocio-pl, do you know where in the code the current zoom level is set or is that done on the websites side?

**Comment by kocio-pl** [comment]
I guess this is done on the side of OSMF, just like metatiles.

**Comment by Adamant36** [comment]
OK. That makes sense.

**Comment by hugobqd** [comment]
Futur is not rendering pixelated map but vector tiles who are theoricaly not limited in zooming.
Z22 example : https://www.maptiler.com/maps/#streets//vector/22/12.33228054/45.43044808

**Comment by kocio-pl** [comment]
So are raster tiles - they have nothing inherent that prohibits it:

https://www.maptiler.com/maps/#streets//raster/22/12.33228054/45.43044808

And we already have ticket for vector migration #3201, but it's not too close by now.

**Comment by matkoniecz** [comment]
> But probably we would like to apply some tweaks when rendering z20.

AFAIK there are some rules already applying to z20, I remember testing road display on z20+ - see form example https://github.com/gravitystorm/openstreetmap-carto/blob/master/roads.mss#L1773 (`[zoom >= 20] { line-width: 24; }`).

**Comment by matkoniecz** [comment]
To be clear - this is something to discuss at https://github.com/openstreetmap/operations/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+z20

But I am guessing that it is blocked by limited capacity of servers.

**Comment by sommerluk** [comment]
> I want to get a starting point and maybe links to other issues in other repositories

> To be clear - this is something to discuss at https://github.com/openstreetmap/operations

Here come some links:

- https://github.com/openstreetmap/openstreetmap-website/issues/2094#issuecomment-450636269 (as per @HolgerJeromin)
- https://github.com/openstreetmap/openstreetmap-website/issues/1672 at the operations side, especially https://github.com/openstreetmap/openstreetmap-website/issues/1672#issuecomment-343158186 and https://github.com/openstreetmap/openstreetmap-website/issues/1672#issuecomment-343166166

In general, [the main issues would be](https://github.com/openstreetmap/openstreetmap-website/issues/1672#issuecomment-343166166)
- Check that the style guys are happy it would render ok
- Get operations to provision sufficient hardware

The second one is clearly out of scope for this repository. But the first one should be in scope. Currently I would not be happy with our z20 rendering. We have very few specific styling for z20. Especially the roads code would need special rules for z20, but also placenames comes in mind. At the current state, we cannot say that we really “support” z20, it's not good enough. I would like to see openstreetmap-carto to support z20 with specific rendering rules.

This is without making any pre-judgement if it is possible to render it later on openstreetmap.org!

> We can think about them if OWG will agree to render z20 first,

While maybe openstreetmap.org is our primary target. But if openstreetmap.org does not render z20,  this does not mean we are forbidden to provide rendering rules for z20 in this style.

**Comment by kocio-pl** [comment]
If you feel it's good to discuss it independent of OWG decision, feel free to reopen the ticket. I told nothing about what is forbidden, I just don't see much value in discussing it without possible delopyment, because this could be true also for any other additional zoom level.

**Comment by matkoniecz** [comment]
>  Especially the roads code would need special rules for z20

This is already done - see https://github.com/gravitystorm/openstreetmap-carto/issues/3686#issuecomment-466654327

Testing z20+ was not the most important part of road redesign but I checked rendering also on higher zoom levels.

Though it is likely that many things can be better, as it is unused and it is reviewed less than other parts of the code.

In general, I see no problems with rendering z20, though at this moment benefits will be visible only in places where there is not enough space to display all icons and on z20 more will be visible.

It is an open question whatever this is a good enough reason to provision necessary hardware and maintain it, but I will leave it to OWG.

**Comment by Adamant36** [comment]
"though at this moment benefits will be visible only in places where there is not enough space to display all icons and on z20 more will be visible."

I could see it being used for other things once it gets implemented. It might be worth brainstorming what exactly before the OWG get involved. 

As far as hardware goes, they will probably need more space for vector maps at some point also wont they? Maybe they could use getting the hardware needed for z20, if there is any, as a way to kill two birds with one stone. Then once there's the space we could put more effort into getting vector rendering done to make it worth it.

**Comment by dktue** [comment]
I think it's easy to over estimate the space needed for z20. Of course, in theory, there are four times more tiles for z20 than for z19. But only a very small amount of them will every be rendered.

Have a look at the [https://wiki.openstreetmap.org/wiki/Tile_disk_usage tile disk usage]. By far less than 1 ‰ of the tiles of z19 have been stored to disk. It will be even less for z20.

**Comment by dktue** [comment]
From @sommerluk 's comment I derive that this ticket has a right to exists as we should make sure that z20 looks good _first_ and _then_ discuss deployement with the hardware guys in the other repository.

I would like to see it reopened regarding the ongoing discussion we're currently seeing here.

**Comment by matkoniecz** [comment]
> we should make sure that z20 looks good _first_

If there are some specific ideas for improvements of z20 - feel free to open new issues.

It would be preferable to discussing all of them in this single issue.

If you think that we are ready for declaring go ahead from side of maintainers of the style (without promising or demanding resources) then it would be probably it would be also be preferable to do it in a new issue.

> this ticket has a right to exists

Yes, but as a closed one - the asked question was answered in https://github.com/gravitystorm/openstreetmap-carto/issues/3686#issuecomment-466755349
